lib/internal/modules/esm/loader.js

```diff
@@ -424,7 +429,45 @@ function createModuleLoader(useCustomLoadersIfPresent = true) {
   return new DefaultModuleLoader();
 }
 
+/**
+ * Get the HooksProxy instance. If it is not defined, then create a new one.
+ * @returns {HooksProxy}
+ */
+function getHooksProxy() {
+  if (!hooksProxy) {
+    const { HooksProxy } = require('internal/modules/esm/hooks');
+    hooksProxy = new HooksProxy();
+  }
+
+  return hooksProxy;
+}
+
+/**
+ * Register a single loader programmatically.
+ * @param {string} specifier
+ * @returns {void}
+ * @example
+ * ```js
+ * register('ts-node/esm');
+ * register('./myLoader.js');
+ * ```
+ */
+function register(specifier) {
+  const { getOptionValue } = require('internal/options');
+  // TODO: Remove this limitation in a follow-up before `register` is released publicly
+  if (getOptionValue('--experimental-loader').length < 1) {
+    const { ERR_ESM_LOADER_REGISTRATION_UNAVAILABLE } = require('internal/errors').codes;
+    throw new ERR_ESM_LOADER_REGISTRATION_UNAVAILABLE();
+  }
+
+  const moduleLoader = require('internal/process/esm_loader').esmLoader;
+
+  moduleLoader.register(specifier);
+}
+
 module.exports = {
-  DefaultModuleLoader,
   createModuleLoader,
+  DefaultModuleLoader,
```

Comment:
Can we keep this in ASCII order?
```suggestion
  DefaultModuleLoader,
  createModuleLoader,
```